public/js/icinga/loader.js

```diff
@@ -828,9 +828,7 @@
             if (!! contentSeparator) {
                 var locationQuery = req.getResponseHeader('X-Icinga-Location-Query');
                 if (locationQuery !== null) {
-                    var a = this.icinga.utils.getUrlHelper().cloneNode(true);
-                    a.search = locationQuery ? '?' + locationQuery : '';
-
+                    let url = currentUrl.path + (locationQuery ? '?' + locationQuery : '');
```

Comment:
Woops, wait a moment, `currentUrl` is defined conditionally. While the block below can make use of it pretty fine (it's guarded by the same condition), this isn't. Parse the url yourself, if `currentUrl` isn't defined yet.

Reply:
ohh yes. But isn't it better if I initialize the `currentUrl` just before (out of) the upper if condition? 🤔

Reply:
Then you'd parse it without needing it in case.

Reply:
We need it for the `url` anyway.